### ISSUE TITLE
Incorrectly detecting error when using $GITHUB_OUTPUT

### ISSUE THREAD

**Issue by Aveline-art** [rhysd/actionlint#291]
Hi. This is a snippet of my Github action.

```yaml
        run: |
          # Build a docker container and
          # push it to ECR so that it can
          # be deployed to ECS.
          cd app
          docker build -f ../stage/Dockerfile -t "$ECR_REGISTRY"/"$ECR_REPOSITORY":"$IMAGE_TAG" .
          docker push "$ECR_REGISTRY"/"$ECR_REPOSITORY":"$IMAGE_TAG"
          echo "image=$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG" >> $GITHUB_OUTPUT
```

When running actionlint via super-linter, I am getting this error

```bash
shellcheck reported issue in this script: SC2086:info:7:58: Double quote to prevent globbing and word splitting [shellcheck]
```

It looks to me that actionlint expects there to be a closing quote on that final echo command. However that is not the [correct syntax](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter) for the new way of setting output. From the docs it looks like [the solution might be to escape the $GITHUB_OUTPUT parameter](https://github.com/rhysd/actionlint/blob/main/docs/checks.md#shellcheck-integration-for-run) since this is how actionlint deals with shellcheck errors.

Do note that super linter is on [v1.6.23](https://github.com/rhysd/actionlint/releases/tag/v1.6.23) for my run, although I did not see this fixed in v.24

**Comment by killermoehre** [comment]
This check just means you want to quote `$GITHUB_OUTPUT`. How should `shellcheck` know that this variable is always on string?

```shell
echo "image=$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG" >> "$GITHUB_OUTPUT"
```

**Comment by rhysd** [comment]
@killermoehre is correct. shellcheck requests to quote the variable.

**Comment by Aveline-art** [comment]
Hmm, so is the recommendation then to escape these in our scripts via enclosing them with quotes? That sounds like it'll work, and I'll give it a go, but I'm not sure that would be friendly to my team. I'll see if I can opt out of this rule.

Thanks for the help guys!

**Comment by killermoehre** [comment]
> Hmm, so is the recommendation then to escape these in our scripts via enclosing them with quotes? That sounds like it'll work, and I'll give it a go, but I'm not sure that would be friendly to my team. I'll see if I can opt out of this rule.
> 
> Thanks for the help guys!

What do you mean by "escape"? It's standard good shell etiquette to quote _every_ variable.

And in shell scripts, quoting ≠ escaping.

**Comment by Aveline-art** [comment]
Ahh I think I see where I got confused. `$GITHUB_OUTPUT` is something native to shell and not Github. Thank you for the helper @killermoehre
